src/components/DatePicker/index.jsx

```diff
@@ -103,7 +103,18 @@ const DatePicker = forwardRef(
       return onChange(allowed, formattedString(allowed, dateFormat));
     };
 
+    const handleOnCalendarChange = date => {
+      if (!date) return;
+
+      const allowed = getAllowedValue(getTimezoneAppliedDateTime(date));
+      setValue(allowed);
+    };
+
     const handleOnOpenChange = open => {
+      if (!open && value) {
+        onChange(value, formattedString(value, dateFormat));
+      }
+
```

Comment:
After first blur action, I guess the `onChange` will keep getting fired after opening and closing the picker, even if values are not changed because the state will retain value. We may have to clear the value in state after sending it to `onChange` on line 115. Please check if this introduces any issues.